File: 2816-lexicographically-smallest-palindrome/2816-lexicographically-smallest-palindrome.py
class Solution:
    def makeSmallestPalindrome(self, s: str) -> str:
        left=0
        right=len(s)-1
        S = list(s)
        if len(s)%2==0:
            while left<right:
                L = S[left]
                R = S[right]
                if L!=R:
                    if L==min(L,R):
                        S[right]=min(L,R)
                    else:
                        S[left]=min(L,R)
                left+=1
                right-=1
        else:
            while left<=(len(s)//2) and right>=((len(s)//2)+1):
                L = S[left]
                R = S[right]
                if L!=R:
                    if L==min(L,R):
                        S[right]=min(L,R)
                    else:
                        S[left]=min(L,R)
                left+=1
                right-=1
        return ''.join(S)

        

File: 2816-lexicographically-smallest-palindrome/2816-lexicographically-smallest-palindrome.js
/**
 * @param {string} s
 * @return {string}
 */
var makeSmallestPalindrome = function(s) {
    let left = 0;
    let right = s.length-1;
    let S = s.split('');

    if (s.length%2===0){
        while (left<right) {
            let L = S[left];
            let R = S[right];

            if (L!==R) {
                let minChar = L<R ? L:R;
                if (L===minChar){
                    S[right]=minChar;
                } else {
                    S[left]=minChar;
                }
            }
            left++;
            right--;
        }
    } else {
        while (left <= Math.floor(s.length/2) && right>=Math.floor(s.length/2)+1) {
            let L = S[left];
            let R = S[right];
            if (L!==R){
                let minChar = L<R ? L:R;
                if (L===minChar) {
                    S[right]=minChar;
                } else {
                    S[left]=minChar;
                }
            }
            left++;
            right--;
        }
    }
    return S.join('');
};